paimon-python/pypaimon/write/ray_datasink.py

```diff
@@ -0,0 +1,85 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+"""
+Module to reawrited a Paimon table from a Ray Dataset, by using the Ray Datasink API.
+"""
+
+from typing import Iterable
+from ray.data.datasource.datasink import Datasink, WriteResult, WriteReturnType
+from pypaimon.table.table import Table
+from pypaimon.write.write_builder import WriteBuilder
+from ray.data.block import BlockAccessor
+from ray.data.block import Block
+from ray.data.dataset import Dataset
+from ray.data._internal.execution.interfaces import TaskContext
+import pyarrow as pa
+
+class PaimonDatasink(Datasink):
+    
+    def __init__(self, table: Table, overwrite=False):
+        self.table = table
+        self.overwrite = overwrite
+
+    def on_write_start(self) -> None:
+        """Callback for when a write job starts.
```

Comment:
<img width="1594" height="522" alt="image" src="https://github.com/user-attachments/assets/62ef517a-6310-4b82-9100-6bedd94103a8" />
 the method change for ray 2.5x